lib/modules/manager/pip-compile/extract.ts

```diff
@@ -97,7 +110,17 @@ export async function extractAllPackageFiles(
       continue;
     }
 
-    for (const packageFile of compileArgs.sourceFiles) {
+    for (const relativeSourceFile of compileArgs.sourceFiles) {
+      const packageFile = upath.normalizeTrim(
+        upath.join(compileDir, relativeSourceFile),
+      );
+      if (packageFile.startsWith('..')) {
```

Comment:
I don't think this is a reliable test

Reply:
Me too, but after normalization it should be enough. If there is a malicious path it should be caught in `readLocalFile` with error. It is either this or `isValidLocalPath` starts working in this case. And I don't know how to do it.

Reply:
`ensureLocalPath` isn't that ugly. And there is only one try..catch now, as only one is needed now.